radix-engine/src/kernel/kernel.rs

```diff
@@ -1242,11 +1268,17 @@ where
 impl<'g, M: KernelCallbackObject, S: CommitableSubstateStore> KernelStackApi for Kernel<'g, M, S> {
     type CallFrameData = M::CallFrameData;
 
-    fn kernel_get_stack_id(&self) -> usize {
-        self.stacks.current_stack
+    fn kernel_get_stack_id(&mut self) -> Result<usize, RuntimeError> {
+        if self.dispatch_get_stack_id_events {
```

Comment:
Maybe worth commenting why only `kernel_get_stack_id()` is conditionally costed.